src/main/java/com/bisnode/opa/client/rest/JsonBodyHandler.java

```diff
@@ -0,0 +1,47 @@
+package com.bisnode.opa.client.rest;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.FieldDefaults;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.net.http.HttpResponse;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+@RequiredArgsConstructor
+@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
+public class JsonBodyHandler<T> implements HttpResponse.BodyHandler<Supplier<T>> {
+
+    JavaType javaType;
```

Comment:
Maybe `ResponseType` would be more appropriate? After all, it's a Java client so what other type could there be?

Reply:
You're right